### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by vsupruniuk**

### CODE REVIEW

**Comment by vsupruniuk** [comment]
image
image
image

File: tests/test_views.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+from taxi.models import Manufacturer, Car
+
+
+class PublicViewsTests(TestCase):
+    def test_index_view_login_required(self) -> None:
+        url = reverse("taxi:index")
+        res = self.client.get(url)
+
+        self.assertNotEqual(res.status_code, 200)
```
Comment (LiudmylaKulinchenko) [inline tests/test_views.py]: You should check if status code is equal to forbidden instead. Now the test will pass even is server error occurred. Same for the next tests

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: tests/test_views.py (hunk 1 above)
Comment (vsupruniuk) [reply tests/test_views.py]: Probably we should check for 302 status code, cause if user unauthorized, django will redirect user to login page with 302 status, also checking for 401/403 doesn't work

Comment (LiudmylaKulinchenko) [reply tests/test_views.py]: I agree!

**Review by LiudmylaKulinchenko — APPROVED** [review]
Well done 🙌

**Pull request closed without merging**